src/globalvars.cpp

```diff
@@ -76,18 +125,137 @@ void update_global_vars(const struct bpf_object *bpf_object,
   }
 
   // Update the values for the global vars (using the above offsets)
-  for (auto [name, offset] : vars_and_offsets) {
-    if (offset < 0) {
-      LOG(BUG) << "Global variable has not been added to the BPF code "
-                  "(codegen_llvm)";
-    }
-
+  for (const auto [global_var, offset] : vars_and_offsets) {
     int64_t *var = reinterpret_cast<int64_t *>(global_vars_buf + offset);
 
-    if (name == NUM_CPUS) {
+    if (global_var == GlobalVar::NUM_CPUS) {
       *var = bpftrace.ncpus_;
+    } else if (global_var == GlobalVar::MAX_CPU_ID) {
+      *var = bpftrace.max_cpu_id_;
```

Comment:
maybe for a follow-up PR but I feel like we should have a map in `bpftrace` of globalVar to these values instead of them being individual member vars.

Reply:
Agreed, the only tough part is read-write vs. read-only needs to be handled differently.

For how, let me change this to a switch statement for clarity as well